packages/storage/src/uploads.ts

```diff
@@ -124,29 +125,53 @@ export const uploadFile = async (
  * (Browser) upload a json reference object via POST to upload service
  * @param ReferenceObject A json reference object to upload
  */
-export const uploadReferenceObject = async (
+export const uploadReference = async (
   referenceObject: ReferenceObject,
 ): Promise<ArweaveResponse> => {
 
   if (Object.keys(referenceObject).length == 0) {
     throw new Error(OBJECT_IS_EMPTY_ERROR);
   }
+  const { media, animation_url, document } = referenceObject;
+  const formdata = new FormData();
 
-  const { media, animation_url } = referenceObject;
-  const canUploadMedia = media?.size < MAX_UPLOAD_BYTES;
-  const canUploadAnimation = animation_url?.size < MAX_UPLOAD_BYTES;
-  
-  if (canUploadMedia) {
-    referenceObject[media] = (await uploadFile(media)).id;
-  }
+  Object.entries(referenceObject).forEach((key: any, value: any): void => {
+    if (key == (media || animation_url || document)) {
+      value?.size < MAX_UPLOAD_BYTES;
+      formdata.append(key, value);
+    } else {
+      formdata.append('field', key, value);
+    }
+  });
 
-  if (canUploadAnimation) {
-    referenceObject[animation_url] = (await uploadFile(animation_url)).id;
-  }
+  try {
+    const request = await fetch(`${ARWEAVE_SERVICE_HOST}/reference`, {
+      method: 'POST',
+      headers: {
+        'mb-api-key': MINTBASE_API_KEY,
```

Comment:
About to be working on what we want to be sending in this header, but probably better to have this working first